core/src/surplus.ts

```diff
@@ -0,0 +1,95 @@
+import type { SurplusAuction, SurplusAuctionUnexistant, SurplusAuctionBaseData } from './types';
+import { getEarliestDate } from './helpers/getEarliestDate';
+import BigNumber from './bignumber';
+import getContract from './contracts';
+import { Contract } from 'ethers';
+
+const surplusAuctionLastIndex = async (contract: Contract) => {
+    const auctionsQuantityBinary = await contract.kicks();
+    return new BigNumber(auctionsQuantityBinary._hex);
+};
+
+const getAuctionState = (earliestEndDate: Date, greatestBid: number) => {
+    const isBidExpired = new Date() > earliestEndDate;
+    const haveBids = greatestBid === 0;
+    if (haveBids) {
+        if (isBidExpired) {
+            return 'ready-for-collection';
+        }
+        return 'have-bids';
+    }
+    if (isBidExpired) {
+        return 'requires-restart';
+    }
+    return 'just-started';
+};
+
+export const fetchSurplusAuctionByIndex = async function (
+    network: string,
+    auctionIndex: number,
+    contractOrNull: Contract | null = null
+): Promise<SurplusAuction | SurplusAuctionUnexistant> {
+    let contract: Contract;
+    if (!!contractOrNull) {
+        contract = contractOrNull;
+    } else {
```

Comment:
- Why do you need to pass contract as a parameter to this function?
- When and why can contract become null?

In short, you don't need to provide it via parameter, please use `getContract` directly

Reply:
if the contract has already been fetched on the previous level ( the function that fetches all the active auction ), then re-getting the contract every time seems inefficient operation. that's why here it's optional and is set if was not provided.

Reply:
still changed the arg defenition to the ts format